### PR TITLE
feat: add commun code for calling ngsild api

### LINKED ISSUE #5: Add common code for calling the ngsil-ld api
- [x] In canteen-alert-manager service
- [x] In csv-ftp-connector service

### PR DESCRIPTION
**Pull request by HoucemKacem**
I'm wondering what is the right approach to test the HTTP calls in entity/subscription services.

### CODE REVIEW

**Comment by bobeal** [comment]
> I'm wondering what is the right approach to test the HTTP calls in entity/subscription services.

you can use WireMock like is done for instance in https://github.com/stellio-hub/stellio-context-broker/blob/develop/subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/NotificationServiceTests.kt

File: lib/build.gradle.kts [hunk 1]
```diff
@@ -4,6 +4,7 @@ plugins {
     id("org.jlleitschuh.gradle.ktlint") version "10.0.0"
     id("io.gitlab.arturbosch.detekt") version "1.16.0"
 
+    kotlin("kapt") version "1.3.61"
```
Comment (bobeal) [inline lib/build.gradle.kts]: 1.4.x ?

File: lib/src/main/kotlin/io/egm/kngsild/api/entities/EntityService.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package io.egm.kngsild.api.entities
```
Comment (bobeal) [inline lib/src/main/kotlin/io/egm/kngsild/api/entities/EntityService.kt]: strange that it is not at the same level than `SubscriptionService`

File: lib/src/main/kotlin/io/egm/kngsild/utils/HttpUtils.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+class HttpUtils {
+
+    val httpClient: HttpClient = HttpClient.newBuilder()
+        .version(HttpClient.Version.HTTP_2)
+        .build()
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    fun getToken(
```
Comment (bobeal) [inline lib/src/main/kotlin/io/egm/kngsild/utils/HttpUtils.kt]: better put it in a `AuthUtils`?

**Comment by HoucemKacem** [comment]
> you can use WireMock like is done for instance in https://github.com/stellio-hub/stellio-context-broker/blob/develop/subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/NotificationServiceTests.kt

@bobeal added.

File: lib/config/detekt/baseline.xml [hunk 4]
```diff
@@ -2,6 +2,8 @@
 <SmellBaseline>
   <ManuallySuppressedIssues></ManuallySuppressedIssues>
   <CurrentIssues>
-    <ID>FunctionOnlyReturningConstant:Library.kt$Library$fun someLibraryMethod(): Boolean</ID>
+    <ID>LongParameterList:EntityService.kt$EntityService$( brokerUrl: String, authServerUrl: String, authClientId: String, authClientSecret: String, authGrantType: String, queryParams: Map&lt;String, String&gt;, contextUrl: String )</ID>
+    <ID>LongParameterList:SubscriptionService.kt$SubscriptionService$( brokerUrl: String, authServerUrl: String, authClientId: String, authClientSecret: String, authGrantType: String, subscriptionPayload: String )</ID>
+    <ID>SwallowedException:UriUtils.kt$catch (e: URISyntaxException) { null }</ID>
```
Comment (bobeal) [inline lib/config/detekt/baseline.xml]: what is this?

File: lib/src/test/kotlin/io/egm/kngsild/api/EntityServiceTest.kt [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            "http://localhost:8090",
+            "client_id",
+            "client_secret",
+            "client_credentials",
+            emptyMap(),
+            coreContext
+        )
+
+        assertTrue(response.isLeft())
+        assertEquals(response, ResourceNotFound("Could not find requested entities").left())
```
Comment (bobeal) [inline lib/src/test/kotlin/io/egm/kngsild/api/EntityServiceTest.kt]: hmm, a query with no result does not return a 404, why do you have this?

File: lib/src/test/kotlin/io/egm/kngsild/utils/AuthUtilsTest.kt [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+            "client_secret",
+            "client_credentials"
+        )
+
+        assertTrue(response.isRight())
+        assertEquals(response, "token".right())
+    }
+
+    @Test
+    fun `it should return AccessTokenNotRetrieved exception if no access token was retrieved`() {
```
Comment (bobeal) [inline lib/src/test/kotlin/io/egm/kngsild/utils/AuthUtilsTest.kt]: it does not "return an exception"

File: lib/config/detekt/baseline.xml (hunk 4 above)
Comment (HoucemKacem) [reply lib/config/detekt/baseline.xml]: It's added in recent versions of Detekt, i'm trying to find how to fix it.
From detekt: `when exceptions are caught and not correctly passed (e.g. as a cause) into a newly thrown exception.  `
I tried to throw a new unparseableException (as done in stellio with a BadRequestDataException) but it stills in the baseline.

File: lib/src/main/kotlin/io/egm/kngsild/utils/HttpUtils.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
+package io.egm.kngsild.utils
+
+import java.net.URLEncoder
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.nio.charset.StandardCharsets
+
+class HttpUtils {
```
Comment (bobeal) [inline lib/src/main/kotlin/io/egm/kngsild/utils/HttpUtils.kt]: couldn't this be an object?

File: lib/config/detekt/baseline.xml (hunk 4 above)
Comment (HoucemKacem) [reply lib/config/detekt/baseline.xml]: @bobeal, I updated the PR, but still have this Exception. 
I added a log when the exception occurs as mentioned [here](https://detekt.github.io/detekt/exceptions.html#compliant-code-4).

Comment (bobeal) [reply lib/config/detekt/baseline.xml]: I fixed it

**Review by bobeal — APPROVED** [review]

**Pull request merged**